auth/cleanup_test.go

```diff
@@ -0,0 +1,120 @@
+package auth
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/traggo/server/model"
+	"github.com/traggo/server/test"
+)
+
+func TestCleanUp_stops(t *testing.T) {
+	db := test.InMemoryDB(t)
+	defer db.Close()
+
+	methodDone := make(chan struct{})
+	stopCleanUp := make(chan bool)
+
+	go func() {
+		CleanUp(db.DB, time.Microsecond*10, stopCleanUp)
+		methodDone <- struct{}{}
+	}()
+
+	stopCleanUp <- true
+
+	select {
+	case <-methodDone:
+		// expected
+	case <-time.After(time.Second):
+		t.Fail()
+	}
+}
+
+func TestCleanUp_removeExpiredDevices(t *testing.T) {
+	now := test.Time("2018-06-30T18:30:00+02:00")
+	timeDispose := fakeTime(now)
+	defer timeDispose()
+
+	db := test.InMemoryDB(t)
+	defer db.Close()
+	db.User(2)
+	db.Create(&model.Device{
+		ID:        1,
+		Token:     "abc",
+		UserID:    2,
+		Name:      "android 1",
+		ExpiresAt: test.Time("2018-07-30T18:30:01+02:00"),
+		ActiveAt:  test.Time("2009-06-30T18:30:00+02:00"),
+		CreatedAt: test.Time("2009-06-30T18:30:00+02:00"),
+	})
+	db.Create(&model.Device{
+		ID:        2,
+		Token:     "abc2",
+		UserID:    2,
+		Name:      "android 2",
+		ExpiresAt: test.Time("2018-06-29T18:30:00+02:00"),
+		ActiveAt:  test.Time("2009-06-30T18:30:00+02:00"),
+		CreatedAt: test.Time("2009-06-30T18:30:00+02:00"),
+	})
+	db.Create(&model.Device{
+		ID:        3,
+		Token:     "abc3",
+		UserID:    2,
+		Name:      "android 3",
+		ExpiresAt: test.Time("2009-06-29T18:30:00+02:00"),
+		ActiveAt:  test.Time("2009-06-30T18:30:00+02:00"),
+		CreatedAt: test.Time("2009-06-30T18:30:00+02:00"),
+	})
+	db.Create(&model.Device{
+		ID:        4,
+		Token:     "abc4",
+		UserID:    2,
+		Name:      "android 4",
+		ExpiresAt: test.Time("2025-06-29T18:30:00+02:00"),
+		ActiveAt:  test.Time("2009-06-30T18:30:00+02:00"),
+		CreatedAt: test.Time("2009-06-30T18:30:00+02:00"),
+	})
+
+	stopCleanUp := make(chan bool)
+
+	go CleanUp(db.DB, time.Microsecond*10, stopCleanUp)
+
+	<-time.After(time.Millisecond * 20)
+
+	expected := []model.Device{
+		{
+			ID:        1,
+			Token:     "abc",
+			UserID:    2,
+			Name:      "android 1",
+			ExpiresAt: test.Time("2018-07-30T18:30:01+02:00"),
+			ActiveAt:  test.Time("2009-06-30T18:30:00+02:00"),
+			CreatedAt: test.Time("2009-06-30T18:30:00+02:00"),
+		},
+		{
+			ID:        4,
+			Token:     "abc4",
+			UserID:    2,
+			Name:      "android 4",
+			ExpiresAt: test.Time("2025-06-29T18:30:00+02:00"),
+			ActiveAt:  test.Time("2009-06-30T18:30:00+02:00"),
+			CreatedAt: test.Time("2009-06-30T18:30:00+02:00"),
+		},
+	}
```

Comment:
Wouldn't that look better in some kind of helper method to improve the readability of the test?
(Especially the four creates before)